### ISSUE TITLE
Water example fails on DX12 on master

### ISSUE THREAD

**Issue by Elabajaba** [gfx-rs/wgpu#2173]
**Description**
The water example fails on the master branch when run with WGPU_BACKEND set to dx12. (the v0.11 branch doesn't crash)

**Repro steps**
Set WGPU_BACKEND to dx12, then run the water example (`cargo run --example water`, release mode also fails)

**Expected vs observed behavior**
Observed: It opens a blank window with a white screen, then crashes instantly.
Expected: It doesn't crash.

**Platform**
WGPU master branch (tested on cf2fb0b665a5dd2e21a1afce817a2f086e56dbcf)
Windows 11, version 10.0.22000 build 22000
AMD 6800xt + 21.11.1 driver

**Extra materials**
```Using AMD Radeon RX 6800 XT (Dx12)
[2021-11-10T20:38:53Z ERROR wgpu::backend::direct] Handling wgpu errors as fatal by default
thread 'main' panicked at 'wgpu error: Validation Error

Caused by:
    In Device::create_render_pipeline
      note: label = `water`
    Internal error in VERTEX shader: D3DCompile error (0x80004005): C:\temp\wgpu\water(87,12-13): error X3003: redefinition of 'x1'


', wgpu\src\backend\direct.rs:2211:5
stack backtrace:
   0: std::panicking::begin_panic_handler
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35\/library\std\src\panicking.rs:517
   1: std::panicking::begin_panic_fmt
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35\/library\std\src\panicking.rs:460
   2: wgpu::backend::direct::default_error_handler
             at .\wgpu\src\backend\direct.rs:2211
   3: core::ops::function::Fn::call<void (*)(enum$<wgpu::Error>),tuple$<enum$<wgpu::Error> > >
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35\library\core\src\ops\function.rs:70
   4: alloc::boxed::impl$46::call<tuple$<enum$<wgpu::Error> >,dyn$<wgpu::UncapturedErrorHandler,assoc$<Output,tuple$<> > > >,alloc::alloc::Global>
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35\library\alloc\src\boxed.rs:1650
   5: wgpu::backend::direct::ErrorSinkRaw::handle_error
             at .\wgpu\src\backend\direct.rs:2199
   6: wgpu::backend::direct::Context::handle_error<enum$<wgpu_core::pipeline::CreateRenderPipelineError> >
             at .\wgpu\src\backend\direct.rs:184
   7: wgpu::backend::direct::impl$3::device_create_render_pipeline
             at .\wgpu\src\backend\direct.rs:1292
   8: wgpu::Device::create_render_pipeline
             at .\wgpu\src\lib.rs:1785
   9: water::impl$1::init
             at .\wgpu\examples\water\main.rs:501
  10: water::framework::start<water::Example>
             at .\wgpu\examples\framework.rs:219
  11: water::framework::run<water::Example>
             at .\wgpu\examples\framework.rs:386
  12: water::main
             at .\wgpu\examples\water\main.rs:790
  13: core::ops::function::FnOnce::call_once<void (*)(),tuple$<> >
             at /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35\library\core\src\ops\function.rs:227
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
error: process didn't exit successfully: `target\debug\examples\water.exe` (exit code: 101)```

**Comment by Elabajaba** [comment]
Tested commits until I found the one where it broke.

 ca7ca749cdb2b5d88512be2b1ec4e44f5a88dfe0 (latest naga version bump) is where it started crashing.

**Comment by kvark** [comment]
Should be fixed by #2172. Please try the latest!

**Comment by Elabajaba** [comment]
Yup, that fixes it.
